extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -171,18 +235,61 @@ export class EntitlementStore {
   }
 
   /**
-   * Returns most qualified platform
-   * @param {!Array<!./entitlement.Entitlement>} entitlements
-   * @returns {!./entitlement.Entitlement}
+   * Returns most qualified platform.
+   * Qualification of a platform is based on an integer weight.
+   * Every platform starts with weight 0 and evaluated against the following parameters,
+   * - user is subscribed with platform (Gives weight 10)
+   * - supports the current viewer (Gives weight 9)
+   *
+   * In the end candidate with max weight is selected.
+   * However if candidate's weight is equal to local platform, then local platform is selected.
+   * @returns {!./subscription-platform.SubscriptionPlatform}
    */
-  selectApplicablePlatform_(entitlements) {
-    let chosenPlatform;
-    entitlements.forEach(platform => {
-      // TODO(@prateekbh): add metering logic here
-      if (platform.enablesThis()) {
-        chosenPlatform = platform;
+  selectApplicablePlatform_() {
+    const localPlatform = this.getLocalPlatform();
+
+    /** @type {!Object<string, number>} */
+    const platformWeights = {};
```

Comment:
Why not just build `weightSorted` array right away? Instead of building a temp map and then copying it over? Further, you keep looking up by serviceId, but you already have platform instances. You can do this:

```
/** @type {!Array<{platform: Platform, weight: number}>} */
const platformWeights = [];
forEach(platform => {
  const weight = calculayeWeight(platform, entitlement);
  platformWeights.push({platform, weight});
});
platformWeights.sort((p1, p2) {
  return p2.weight - p1.weight;
});
// the rest as you have
```